### DISCUSSION TITLE
OpenTelemetry metrics and traces

### DISCUSSION THREAD

**Discussion by honestegg** [datalust/seq-tickets#1972] [Help] [answered]
Does Seq support OpenTelemetry metrics and traces? Or just logs? The documentation mentions the trace endpoint:
https://docs.datalust.co/docs/server-http-api#ingest

I haven't got it to work so far -- I'm just getting a 404 response from Seq.

**Comment by nblumhardt** [comment, marked as answer]
Sorry about the confusion, @honestegg - Seq doesn't currently support traces, but we're working on an implementation right now, and the trace ingestion endpoint does (partially) exist in the shipped product, but behind a feature flag.

I missed the documentation generator when implementing the feature flagging mechanism, we'll get this fixed 😅 

We'll post some more details on our progress soon. (If you're interested in following development more closely, please drop our `support@` address an email, we're setting up a community Slack and would love to invite you.)

**Reply by robv8r** [reply]
@nblumhardt, Thank you so much for integrating Open Telemetry into Seq.  Do you happen to have a timeframe for Traces?

**Reply by nblumhardt** [reply]
Hi @robv8r! Good news, traces [shipped early last year](https://blog.datalust.co/seq-2024-1-released/), have been through a couple of rounds of improvements, and are now super-nice 😎

**Reply by liammclennan** [reply]
Hi @robv8r & @honestegg ,

Seq supports logs and traces from 2024.1 - see https://blog.datalust.co/seq-2024-1-released/ 

OpenTelemetry is supported via gRPC or HttpProtobuf.

As of 2024.3, Seq does not yet support OpenTelemetry metrics. It is listed on the [roadmap](https://datalust.co/roadmap) but we do not have a time frame.

**Reply by Edgaras91** [reply]
Hey @liammclennan, I wonder if metrics is behind some sort of feature flag now? If not, maybe a rough estimate of when that would be the case?

**Reply by nblumhardt** [reply]
Hi @Edgaras91,

Metrics will ship in Seq 2026.1; we're aiming for early 2026, but we're making a deep investment in optimized, columnar storage, which is a complex project, so there won't be a precise ship date for some time yet.

We're hoping to have an early preview out later this year - I've just created https://github.com/datalust/seq-tickets/issues/2491 so you can subscribe to that thread and receive updates. Cheers!

**Reply by liammclennan** [reply]
@honestegg @Edgaras91 @robv8r 

Metrics (plus lots of other things) has [shipped in 2026.1](https://datalust.co/blog/seq-2026-1-released).
